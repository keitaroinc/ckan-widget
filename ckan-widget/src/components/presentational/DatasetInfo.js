import React, { Component } from 'react'
import DatasetDetails from '../presentational/DatasetDetails'


class DatasetInfo extends Component{
    constructor(props) {
        super(props)
        this.state = { collapsed: true }
    }

    findFormats = resources => {
        let formats = []

        if (Array.isArray(resources) && resources.length){
            resources.forEach( resource => {
                const format = resource.format
                if (!formats.includes(format)) {
                    formats.push(format)
                }
            })
        }

        return formats
    }

    formatDate = date => {
        const d = new Date(date )
        return d.toUTCString()
    }

    handleDatasetClick = () => {
        this.setState({ collapsed: !this.state.collapsed })
    }

    render(){
        const { title, notes, metadata_modified, datatype, resources } = this.props
        const datetime = this.formatDate(metadata_modified)
        const formats = this.findFormats(resources)
        const collapseClass = this.state.collapsed ? '' : 'collapsed'

        return(
            <div className={"card my-3 " + collapseClass}>
                <div className="dataset card-body" onClick={this.handleDatasetClick}>
                    <i className="material-icons">expand_more</i>
                    <div className="row">
                        <div className="col-lg-3 d-flex justify-content-lg-center align-items-center mb-lg-0 mb-md-3">
                            <img className="img-thumbnail img-fluid" src="https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg" alt="logo" />
                        </div>
                        <div className="col-lg-9">
                            <h3 className="title">{title}</h3>
                            <p className="text-muted">{notes && notes.length > 130 ? `${notes.substring(0, 130)}...` : notes}</p>
                            <ul className="list-inline">
                              <li className="list-inline-item">Modified: {datetime}</li>
                              <li className="list-inline-item">Formats: {formats !== undefined ? formats.join(', ') : 'N/A'}</li>
                              <li className="list-inline-item">Datatype: {datatype !== undefined ? datatype.join(', ') : 'N/A'}</li>
                            </ul>
                        </div>
                    </div>
                </div>
                <DatasetDetails collapsed={this.state.collapsed} {...this.props} />
            </div>
        )
    }
}
export default DatasetInfo;
