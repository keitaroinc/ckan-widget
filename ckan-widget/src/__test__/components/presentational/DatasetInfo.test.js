import React from 'react';
import { shallow } from 'enzyme';
import DatasetInfo from '../../../components/presentational/DatasetInfo';

let component

describe('DatasetInfo', () => {
  beforeEach(() => {
    component = shallow(
      <DatasetInfo
        title={'Dataset title'}
        notes={'Describe dataset'}
        metadata_modified={'2018-08-23T07:26:42.122834'}
        resources={[{ restricted: "{\"level\": \"public\"}", format: 'CSV', name: 'rsc1', last_modified: '21 June 2018' }]} />
    )
  })

  it('should be defined', () => {
    expect(DatasetInfo).toBeDefined();
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should receive props', () => {
    expect(component.instance().props.title).toBe('Dataset title')
    expect(component.instance().props.notes).toBe('Describe dataset')
    expect(component.instance().props.metadata_modified).toBe('2018-08-23T07:26:42.122834')
    expect(component.instance().props.resources.length).toEqual(1)
  });

  it('should format data properly', () => {
    expect(component.instance().formatDate('2018-08-23T07:26:42.122834')).toBe('Thu, 23 Aug 2018 05:26:42 GMT')
  })

  it('should find formats of the resources', () => {
    const resources = [
      { restricted: "{\"level\": \"public\"}", format: 'CSV', name: 'rsc1', last_modified: '21 June 2018' },
      { restricted: "{\"level\": \"public\"}", format: 'XML', name: 'rsc2', last_modified: '25 June 2018' }
    ]

    const formats = component.instance().findFormats(resources)
    expect(formats.length).toEqual(2)
    expect(formats).toEqual(['CSV', 'XML'])
  })

  it('should expand on click', () => {
    expect(component.instance().state.collapsed).toBe(true)

    const dataset = component.find('div.dataset')
    dataset.simulate('click')
    expect(component.instance().state.collapsed).toBe(false)
  })
});