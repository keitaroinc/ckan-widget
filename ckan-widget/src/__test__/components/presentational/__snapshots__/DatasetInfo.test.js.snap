// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetInfo should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatasetInfo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="dataset card-body"
          onClick={[Function]}
>
          <i
                    className="material-icons"
          >
                    expand_more
          </i>
          <div
                    className="row"
          >
                    <div
                              className="col-lg-3 d-flex justify-content-lg-center align-items-center mb-lg-0 mb-md-3"
                    >
                              <img
                                        alt="logo"
                                        className="img-thumbnail img-fluid"
                                        src="https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg"
                              />
                    </div>
                    <div
                              className="col-lg-9"
                    >
                              <h4
                                        className="title text-primary"
                              />
                              <p
                                        className="text-muted"
                              />
                              <ul
                                        className="list-inline"
                              >
                                        <li
                                                  className="list-inline-item"
                                        >
                                                  <strong>
                                                            Modified:
                                                  </strong>
                                                   
                                                  Invalid Date
                                        </li>
                                        <li
                                                  className="list-inline-item"
                                        >
                                                  <strong>
                                                            Formats:
                                                  </strong>
                                                   
                                                  
                                        </li>
                                        <li
                                                  className="list-inline-item"
                                        >
                                                  <strong>
                                                            Datatype:
                                                  </strong>
                                                   
                                                  N/A
                                        </li>
                              </ul>
                    </div>
          </div>
</div>,
        <DatasetDetails
          collapsed={true}
/>,
      ],
      "className": "card dataset-wrap my-3 ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="material-icons"
>
              expand_more
</i>,
            <div
              className="row"
>
              <div
                            className="col-lg-3 d-flex justify-content-lg-center align-items-center mb-lg-0 mb-md-3"
              >
                            <img
                                          alt="logo"
                                          className="img-thumbnail img-fluid"
                                          src="https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg"
                            />
              </div>
              <div
                            className="col-lg-9"
              >
                            <h4
                                          className="title text-primary"
                            />
                            <p
                                          className="text-muted"
                            />
                            <ul
                                          className="list-inline"
                            >
                                          <li
                                                        className="list-inline-item"
                                          >
                                                        <strong>
                                                                      Modified:
                                                        </strong>
                                                         
                                                        Invalid Date
                                          </li>
                                          <li
                                                        className="list-inline-item"
                                          >
                                                        <strong>
                                                                      Formats:
                                                        </strong>
                                                         
                                                        
                                          </li>
                                          <li
                                                        className="list-inline-item"
                                          >
                                                        <strong>
                                                                      Datatype:
                                                        </strong>
                                                         
                                                        N/A
                                          </li>
                            </ul>
              </div>
</div>,
          ],
          "className": "dataset card-body",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "expand_more",
              "className": "material-icons",
            },
            "ref": null,
            "rendered": "expand_more",
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-lg-3 d-flex justify-content-lg-center align-items-center mb-lg-0 mb-md-3"
>
                  <img
                                    alt="logo"
                                    className="img-thumbnail img-fluid"
                                    src="https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg"
                  />
</div>,
                <div
                  className="col-lg-9"
>
                  <h4
                                    className="title text-primary"
                  />
                  <p
                                    className="text-muted"
                  />
                  <ul
                                    className="list-inline"
                  >
                                    <li
                                                      className="list-inline-item"
                                    >
                                                      <strong>
                                                                        Modified:
                                                      </strong>
                                                       
                                                      Invalid Date
                                    </li>
                                    <li
                                                      className="list-inline-item"
                                    >
                                                      <strong>
                                                                        Formats:
                                                      </strong>
                                                       
                                                      
                                    </li>
                                    <li
                                                      className="list-inline-item"
                                    >
                                                      <strong>
                                                                        Datatype:
                                                      </strong>
                                                       
                                                      N/A
                                    </li>
                  </ul>
</div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="logo"
                    className="img-thumbnail img-fluid"
                    src="https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg"
/>,
                  "className": "col-lg-3 d-flex justify-content-lg-center align-items-center mb-lg-0 mb-md-3",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "logo",
                    "className": "img-thumbnail img-fluid",
                    "src": "https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4
                      className="title text-primary"
/>,
                    <p
                      className="text-muted"
/>,
                    <ul
                      className="list-inline"
>
                      <li
                                            className="list-inline-item"
                      >
                                            <strong>
                                                                  Modified:
                                            </strong>
                                             
                                            Invalid Date
                      </li>
                      <li
                                            className="list-inline-item"
                      >
                                            <strong>
                                                                  Formats:
                                            </strong>
                                             
                                            
                      </li>
                      <li
                                            className="list-inline-item"
                      >
                                            <strong>
                                                                  Datatype:
                                            </strong>
                                             
                                            N/A
                      </li>
</ul>,
                  ],
                  "className": "col-lg-9",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "title text-primary",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "text-muted",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <li
                          className="list-inline-item"
>
                          <strong>
                                                    Modified:
                          </strong>
                           
                          Invalid Date
</li>,
                        <li
                          className="list-inline-item"
>
                          <strong>
                                                    Formats:
                          </strong>
                           
                          
</li>,
                        <li
                          className="list-inline-item"
>
                          <strong>
                                                    Datatype:
                          </strong>
                           
                          N/A
</li>,
                      ],
                      "className": "list-inline",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <strong>
                              Modified:
</strong>,
                            " ",
                            "Invalid Date",
                          ],
                          "className": "list-inline-item",
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Modified:",
                            },
                            "ref": null,
                            "rendered": "Modified:",
                            "type": "strong",
                          },
                          " ",
                          "Invalid Date",
                        ],
                        "type": "li",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <strong>
                              Formats:
</strong>,
                            " ",
                            "",
                          ],
                          "className": "list-inline-item",
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Formats:",
                            },
                            "ref": null,
                            "rendered": "Formats:",
                            "type": "strong",
                          },
                          " ",
                          "",
                        ],
                        "type": "li",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <strong>
                              Datatype:
</strong>,
                            " ",
                            "N/A",
                          ],
                          "className": "list-inline-item",
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Datatype:",
                            },
                            "ref": null,
                            "rendered": "Datatype:",
                            "type": "strong",
                          },
                          " ",
                          "N/A",
                        ],
                        "type": "li",
                      },
                    ],
                    "type": "ul",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "collapsed": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dataset card-body"
            onClick={[Function]}
>
            <i
                        className="material-icons"
            >
                        expand_more
            </i>
            <div
                        className="row"
            >
                        <div
                                    className="col-lg-3 d-flex justify-content-lg-center align-items-center mb-lg-0 mb-md-3"
                        >
                                    <img
                                                alt="logo"
                                                className="img-thumbnail img-fluid"
                                                src="https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg"
                                    />
                        </div>
                        <div
                                    className="col-lg-9"
                        >
                                    <h4
                                                className="title text-primary"
                                    />
                                    <p
                                                className="text-muted"
                                    />
                                    <ul
                                                className="list-inline"
                                    >
                                                <li
                                                            className="list-inline-item"
                                                >
                                                            <strong>
                                                                        Modified:
                                                            </strong>
                                                             
                                                            Invalid Date
                                                </li>
                                                <li
                                                            className="list-inline-item"
                                                >
                                                            <strong>
                                                                        Formats:
                                                            </strong>
                                                             
                                                            
                                                </li>
                                                <li
                                                            className="list-inline-item"
                                                >
                                                            <strong>
                                                                        Datatype:
                                                            </strong>
                                                             
                                                            N/A
                                                </li>
                                    </ul>
                        </div>
            </div>
</div>,
          <DatasetDetails
            collapsed={true}
/>,
        ],
        "className": "card dataset-wrap my-3 ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="material-icons"
>
                expand_more
</i>,
              <div
                className="row"
>
                <div
                                className="col-lg-3 d-flex justify-content-lg-center align-items-center mb-lg-0 mb-md-3"
                >
                                <img
                                                alt="logo"
                                                className="img-thumbnail img-fluid"
                                                src="https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg"
                                />
                </div>
                <div
                                className="col-lg-9"
                >
                                <h4
                                                className="title text-primary"
                                />
                                <p
                                                className="text-muted"
                                />
                                <ul
                                                className="list-inline"
                                >
                                                <li
                                                                className="list-inline-item"
                                                >
                                                                <strong>
                                                                                Modified:
                                                                </strong>
                                                                 
                                                                Invalid Date
                                                </li>
                                                <li
                                                                className="list-inline-item"
                                                >
                                                                <strong>
                                                                                Formats:
                                                                </strong>
                                                                 
                                                                
                                                </li>
                                                <li
                                                                className="list-inline-item"
                                                >
                                                                <strong>
                                                                                Datatype:
                                                                </strong>
                                                                 
                                                                N/A
                                                </li>
                                </ul>
                </div>
</div>,
            ],
            "className": "dataset card-body",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "expand_more",
                "className": "material-icons",
              },
              "ref": null,
              "rendered": "expand_more",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="col-lg-3 d-flex justify-content-lg-center align-items-center mb-lg-0 mb-md-3"
>
                    <img
                                        alt="logo"
                                        className="img-thumbnail img-fluid"
                                        src="https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg"
                    />
</div>,
                  <div
                    className="col-lg-9"
>
                    <h4
                                        className="title text-primary"
                    />
                    <p
                                        className="text-muted"
                    />
                    <ul
                                        className="list-inline"
                    >
                                        <li
                                                            className="list-inline-item"
                                        >
                                                            <strong>
                                                                                Modified:
                                                            </strong>
                                                             
                                                            Invalid Date
                                        </li>
                                        <li
                                                            className="list-inline-item"
                                        >
                                                            <strong>
                                                                                Formats:
                                                            </strong>
                                                             
                                                            
                                        </li>
                                        <li
                                                            className="list-inline-item"
                                        >
                                                            <strong>
                                                                                Datatype:
                                                            </strong>
                                                             
                                                            N/A
                                        </li>
                    </ul>
</div>,
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt="logo"
                      className="img-thumbnail img-fluid"
                      src="https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg"
/>,
                    "className": "col-lg-3 d-flex justify-content-lg-center align-items-center mb-lg-0 mb-md-3",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "logo",
                      "className": "img-thumbnail img-fluid",
                      "src": "https://www.datasud.fr/wp-content/themes/crigepaca/assets/images/logo_region_paca.jpg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h4
                        className="title text-primary"
/>,
                      <p
                        className="text-muted"
/>,
                      <ul
                        className="list-inline"
>
                        <li
                                                className="list-inline-item"
                        >
                                                <strong>
                                                                        Modified:
                                                </strong>
                                                 
                                                Invalid Date
                        </li>
                        <li
                                                className="list-inline-item"
                        >
                                                <strong>
                                                                        Formats:
                                                </strong>
                                                 
                                                
                        </li>
                        <li
                                                className="list-inline-item"
                        >
                                                <strong>
                                                                        Datatype:
                                                </strong>
                                                 
                                                N/A
                        </li>
</ul>,
                    ],
                    "className": "col-lg-9",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "className": "title text-primary",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "h4",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "className": "text-muted",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <li
                            className="list-inline-item"
>
                            <strong>
                                                        Modified:
                            </strong>
                             
                            Invalid Date
</li>,
                          <li
                            className="list-inline-item"
>
                            <strong>
                                                        Formats:
                            </strong>
                             
                            
</li>,
                          <li
                            className="list-inline-item"
>
                            <strong>
                                                        Datatype:
                            </strong>
                             
                            N/A
</li>,
                        ],
                        "className": "list-inline",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              <strong>
                                Modified:
</strong>,
                              " ",
                              "Invalid Date",
                            ],
                            "className": "list-inline-item",
                          },
                          "ref": null,
                          "rendered": Array [
                            Object {
                              "instance": null,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "children": "Modified:",
                              },
                              "ref": null,
                              "rendered": "Modified:",
                              "type": "strong",
                            },
                            " ",
                            "Invalid Date",
                          ],
                          "type": "li",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              <strong>
                                Formats:
</strong>,
                              " ",
                              "",
                            ],
                            "className": "list-inline-item",
                          },
                          "ref": null,
                          "rendered": Array [
                            Object {
                              "instance": null,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "children": "Formats:",
                              },
                              "ref": null,
                              "rendered": "Formats:",
                              "type": "strong",
                            },
                            " ",
                            "",
                          ],
                          "type": "li",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              <strong>
                                Datatype:
</strong>,
                              " ",
                              "N/A",
                            ],
                            "className": "list-inline-item",
                          },
                          "ref": null,
                          "rendered": Array [
                            Object {
                              "instance": null,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "children": "Datatype:",
                              },
                              "ref": null,
                              "rendered": "Datatype:",
                              "type": "strong",
                            },
                            " ",
                            "N/A",
                          ],
                          "type": "li",
                        },
                      ],
                      "type": "ul",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "collapsed": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
